File: market/test/java/ru/yandex/market/routing/quartz/TmsTaskTest.java
package ru.yandex.market.routing.quartz;

import org.junit.jupiter.api.Test;
import org.mockito.Mockito;
import org.quartz.JobDetail;
import org.quartz.JobExecutionContext;
import org.quartz.JobKey;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
import org.springframework.test.context.ContextConfiguration;

import ru.yandex.market.routing.AbstractFunctionalTest;
import ru.yandex.market.routing.quartz.task.TmsTasks;

import ru.yandex.market.javaframework.quartz.test.AbstractQuartzTest;
import ru.yandex.market.tms.quartz2.model.Executor;

public class TmsTaskTest extends AbstractFunctionalTest {

    @Autowired
    public NamedParameterJdbcTemplate jdbcTemplate;

    @Autowired
    @Qualifier("tmsLogsCleanupExecutor")
    public Executor tmsLogsCleanupExecutor;

    protected JobExecutionContext mockContext() {
        final JobExecutionContext mockContext = Mockito.mock(JobExecutionContext.class);
        final JobDetail mockJobDetail = Mockito.mock(JobDetail.class);
        Mockito.when(mockContext.getJobDetail()).thenReturn(mockJobDetail);
        Mockito.when(mockJobDetail.getKey()).thenReturn(new JobKey("test"));
        return mockContext;
    }

    @Test
    public void testTask() {
        tmsLogsCleanupExecutor.doJob(mockContext());
        jdbcTemplate.query("select * from tms.qrtz_job_details", (rs) -> {
            System.out.println(rs.getString("JOB_NAME"));
        });
    }

}

File: market/test/kotlin/ru/yandex/market/partner/status/quartz/TmsTaskTest.kt
package ru.yandex.market.partner.status.quartz

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import ru.yandex.market.common.test.db.DbUnitDataSet
import ru.yandex.market.partner.status.AbstractFunctionalTest
import ru.yandex.market.tms.quartz2.model.Executor

class TmsTaskTest : AbstractFunctionalTest() {

    @Autowired
    private lateinit var tmsLogsCleanupExecutor: Executor

    @Test
    @DbUnitDataSet(
        before = ["TmsTaskTest/oldLogs.before.csv"],
        after = ["TmsTaskTest/oldLogs.after.csv"]
    )
    fun testTask() {
        tmsLogsCleanupExecutor.doJob(null)
    }
}
